fvm/src/executor/default.rs

```diff
@@ -66,46 +68,62 @@ where
-                machine,
-                msg.gas_limit,
-                sender_id,
-                msg.sequence,
-                msg.gas_premium.clone(),
-            );
-            // This error is fatal because it should have already been accounted for inside
-            // preflight_message.
-            if let Err(e) = cm.charge_gas(inclusion_cost) {
-                return (Err(e), cm.finish().1);
-            }
-
-            let params = if msg.params.is_empty() {
-                None
-            } else {
-                Some(Block::new(DAG_CBOR, msg.params.bytes()))
-            };
+        let (res, gas_used, mut backtrace, exec_trace, events_root, events) =
+            self.map_machine(|machine| {
+                // We're processing a chain message, so the sender is the origin of the call stack.
+                let mut cm = K::CallManager::new(
+                    machine,
+                    msg.gas_limit,
+                    sender_id,
+                    msg.sequence,
+                    msg.gas_premium.clone(),
+                );
+                // This error is fatal because it should have already been accounted for inside
+                // preflight_message.
+                if let Err(e) = cm.charge_gas(inclusion_cost) {
+                    return (Err(e), cm.finish().1);
+                }
 
-            let result = cm.with_transaction(|cm| {
-                // Invoke the message.
-                let ret = cm.send::<K>(sender_id, msg.to, msg.method_num, params, &msg.value)?;
+                let params = if msg.params.is_empty() {
+                    None
+                } else {
+                    Some(Block::new(DAG_CBOR, msg.params.bytes()))
+                };
 
-                // Charge for including the result (before we end the transaction).
-                if let InvocationResult::Return(value) = &ret {
-                    cm.charge_gas(cm.context().price_list.on_chain_return_value(
-                        value.as_ref().map(|v| v.size() as usize).unwrap_or(0),
-                    ))?;
-                }
+                let result = cm.with_transaction(|cm| {
+                    // Invoke the message.
+                    let ret =
+                        cm.send::<K>(sender_id, msg.to, msg.method_num, params, &msg.value)?;
+
+                    // Charge for including the result (before we end the transaction).
+                    if let InvocationResult::Return(value) = &ret {
+                        cm.charge_gas(cm.context().price_list.on_chain_return_value(
+                            value.as_ref().map(|v| v.size() as usize).unwrap_or(0),
+                        ))?;
+                    }
+
+                    Ok(ret)
+                });
+                let (res, machine) = cm.finish();
+
+                // Flush all events to the store.
+                let events_root = match machine.commit_events(res.events.as_slice()) {
```

Comment:
All of the above is noise due to a newline introduced by `cargo fmt`. Here is where the change actually starts.